Fix build on 10.11 and earlier by not using clock_gettime_nsec_np.
https://bugs.php.net/bug.php?id=81137
https://github.com/php/php-src/commit/72c0deb4009e1ddfdccd2b353e0a839ffe10d81a
--- a/ext/standard/hrtime.c
+++ b/ext/standard/hrtime.c
@@ -34,6 +34,12 @@
 
 static double _timer_scale = .0;
 
+#elif PHP_HRTIME_PLATFORM_APPLE
+
+# include <mach/mach_time.h>
+# include <string.h>
+static mach_timebase_info_data_t _timerlib_info;
+
 #elif PHP_HRTIME_PLATFORM_HPUX
 
 # include <sys/time.h>
@@ -60,7 +66,9 @@ static int _timer_init(void)
 
 #elif PHP_HRTIME_PLATFORM_APPLE
 
-	/* pass */
+	if (mach_timebase_info(&_timerlib_info)) {
+		return -1;
+	}
 
 #elif PHP_HRTIME_PLATFORM_POSIX
 
@@ -107,8 +115,7 @@ static zend_always_inline php_hrtime_t _timer_current(void)
 	QueryPerformanceCounter(&lt);
 	return (php_hrtime_t)((php_hrtime_t)lt.QuadPart * _timer_scale);
 #elif PHP_HRTIME_PLATFORM_APPLE
-	/* the value is of php_hrtime_t type already */
-	return clock_gettime_nsec_np(CLOCK_MONOTONIC_RAW);
+	return (php_hrtime_t)mach_absolute_time() * _timerlib_info.numer / _timerlib_info.denom;
 #elif PHP_HRTIME_PLATFORM_POSIX
 	struct timespec ts = { .tv_sec = 0, .tv_nsec = 0 };
 	if (0 == clock_gettime(CLOCK_MONOTONIC, &ts)) {
